libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/KeyValueEncryptedFileStore.java

```diff
@@ -0,0 +1,195 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.androidsdk.smartstore.store;
+
+import android.text.TextUtils;
+import com.salesforce.androidsdk.analytics.security.DecrypterInputStream;
+import com.salesforce.androidsdk.analytics.security.EncrypterOutputStream;
+import com.salesforce.androidsdk.security.SalesforceKeyGenerator;
+import com.salesforce.androidsdk.smartstore.util.SmartStoreLogger;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+/** Key-value store backed by file system */
+public class KeyValueEncryptedFileStore  {
+
+    private static final String TAG = KeyValueEncryptedFileStore.class.getSimpleName();
+    private final String encryptionKey;
+    private final File storeDir;
+
+    /**
+     * Constructor
+     *
+     * @param parentDir parent directory for key value store
+     * @param storeName name for key value store
+     * @param encryptionKey encryption key for key value store
+     */
+    public KeyValueEncryptedFileStore(File parentDir, String storeName, String encryptionKey) {
+        storeDir = new File(parentDir, storeName);
+        if (!storeDir.exists()) {
+            storeDir.mkdirs();
+        }
+        this.encryptionKey = encryptionKey;
+    }
+
+    /**
+     * Save value for the given key.
+     *
+     * @param key Unique identifier.
+     * @param value Value to be persisted.
+     * @return True - if successful, False - otherwise.
+     */
+    public boolean saveValue(String key, String value) {
+        if (!isKeyValid(key, "saveValue")) {
+            return false;
+        }
+        if (value == null) {
+            SmartStoreLogger.w(TAG, "saveValue: Invalid value supplied: " + key);
```

Comment:
Shouldn't this be `value`, in this log line? Also, is empty string a valid `value`? If not, `TextUtils.isEmpty(value)` might be a better check.